File: trainInfo.php
<?php
$row = 1;
$result = array();
if (($handle = fopen("test1.csv", "r")) != FALSE) {
	// Ignore the header of the csv file
	fgets($handle);
	// Read data and store it in a 2-d array
	while (($data = fgetcsv($handle, 100, ","))!== FALSE ){
		$num = count($data);
		$result[$row -1] = $data;
		$row++;	
	}

	// Foreach and array_multisort used to sort data based on "route_id"
	foreach ($result as $key => $row){
		$runId[$key] = $row[2];
	}
	array_multisort($runId, SORT_ASC, $result); 
	fclose($handle);

	// Output the result in a human-friendly format
	echo "Train Line\tRoute\t\tRun Number\tOperator ID\n";
	for($i = 0; $i < sizeof($result); $i ++){
		echo $result[$i][0]."\t\t".$result[$i][1]."\t".$result[$i][2]."\t\t".$result[$i][3]."\n" ;
	}	
}
?>

File: trainInfo.rb
#!/usr/bin/ruby
require 'csv'
#result = CSV.read("test1.csv")
#csv_text = File.read("test1.csv")
#csv = CSV.parse(csv_text, :headers=> false)
#result = csv[1..-1]
#puts result[1]

arr = []
CSV.foreach("test1.csv") do |row|
	arr << row
end
arr = arr[1..-1]
result = arr.sort {|a,b| a[2] <=> b[2]}
puts "Train Line\tRoute\t\tRun Number\tOperator ID"
for i in 0..result.count-1
puts result[i][0]+"\t\t"+result[i][1]+"\t"+result[i][2]+"\t\t"+result[i][3]
end
